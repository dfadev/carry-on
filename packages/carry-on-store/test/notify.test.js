import notifySubscribers from "../src/notify";

test("notify match", () => {
  expect(notifySubscribers()).toMatchSnapshot();
});

test("plug snapshot", () => {
  const plug = {};
  let state = { plug };
  const query = () => state;
  const dispatch = (action, type, force, ...args) => {
    return (state = action(state));
  };

  const notify = notifySubscribers();
  expect(notify).toMatchSnapshot();
});

test("subscribe/unsubscribe", () => {
  const plug = {};
  let state = { plug };
  const query = () => state;
  let dispatch = (action, type, force, ...args) => {
    return (state = action(state));
  };

  const notify = notifySubscribers();
  expect(notify).toMatchSnapshot();

  dispatch = notify.plugin.dispatch({ dispatch });

  let msgRecvCount = 0;
  const fn = state => { msgRecvCount++; };
  const unsubscribe = notify.subscribe(fn);
  expect(notify.subscribers.length).toEqual(1);

  dispatch(state => state);
  expect(msgRecvCount).toBe(1);

  unsubscribe();
  expect(notify.subscribers.length).toEqual(0);

  dispatch(state => state);
  expect(msgRecvCount).toBe(1);
});
