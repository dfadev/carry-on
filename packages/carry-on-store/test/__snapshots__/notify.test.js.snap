// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`notify match 1`] = `
Object {
  "plugin": Object {
    "dispose": [Function],
    "middleware": [Function],
  },
  "subscribe": [Function],
  "subscribers": Map {},
}
`;

exports[`plug snapshot 1`] = `
Object {
  "plugin": Object {
    "dispose": [Function],
    "middleware": [Function],
  },
  "subscribe": [Function],
  "subscribers": Map {},
}
`;

exports[`subscribe/unsubscribe 1`] = `
Object {
  "plugin": Object {
    "dispose": [Function],
    "middleware": [Function],
  },
  "subscribe": [Function],
  "subscribers": Map {},
}
`;
